boards/arm/stm32h747i_disco/Kconfig.defconfig

```diff
@@ -0,0 +1,27 @@
+# Kconfig - STM32H747I DISCOVERY board configuration
+#
+# Copyright (c) 2019 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if BOARD_STM32H747I_DISCO_M7
+
+config BOARD
+	default "stm32h747i_disco_m7" if BOARD_STM32H747I_DISCO_M7
+
+if UART_CONSOLE
+
+config UART_1
+	default y if BOARD_STM32H747I_DISCO_M7
```

Comment:
As peripherals are shared between cortex contexts i would set configs to n by default for peripheral to ensure that is not used by M4 and M7

Reply:
Problem is that I need one UART defined on one side to enable serial testing (and things that require serial).

But, I agree with your concern and this is actually why I preferred to put devices activation in this file common to both cores rather than in _m{4.7}_defconfig files.
What I can add is the following to make the intention clearer (won't add nothing in term of "mechanical" check though :
```
config UART_1
	default y if BOARD_STM32H747I_DISCO_M7
	default n if BOARD_STM32H747I_DISCO_M4
```
And the same should be done for each other peripheral device.

Reply:
ok for me